tools/autograd/templates/Functions.cpp

```diff
@@ -694,31 +694,24 @@ Tensor masked_scatter_backward(const Tensor & grad, const Tensor & mask, IntArra
 
 Tensor cholesky_backward(Tensor grad, bool upper, Tensor L) {
   // cf. Iain Murray (2016); arXiv 1602.07527
+  // This gradient is symmetric, and not triangular.
+  // This choice is due to the ambiguity caused by taking triangular gradients,
+  // described in the example below:
+  // Consider a matrix A which is PD, and compute the trace of A's inverse using
+  // A.inverse().trace(). The gradient of this trace w.r.t to A in this case is symmetric
+  // and unique. Note that A = L @ L^{T} => inv(A) = inv(L)^{T} @ inv(L)
```

Comment:
I think the root problem isn't about this specific case, but about the assumption made by `cholesky`, i.e., the input being symmetric. It is because of this that the gradient can be freely distributed for each reflected pair of values. The motivation, I think, should be that symmetric gradient leads to (usually) more stable gradient updates, and (if the matrix is updated with SGD-like algorithm) leaves the matrix hermitian. What do you think?

Reply:
I think this sounds good. I wasn't sure if a motivational statement should be presented here, which is why refrained from adding it.

But yes, I should have elucidated more about the root problem. Please allow me to make changes to the note.

Reply:
I think this is where we make a design choice. And a good explanation of the ambiguity and the reason we make the choice will be very useful. :)

Reply:
I've edited the note, thanks for the suggestions.